### ISSUE TITLE
Fire Charge Crash

### ISSUE THREAD

**Issue by RavenHwk** [Alex-the-666/Ice_and_Fire#3425]
Hullo! I looked through other crashes but haven't found anything quite like mine. I was fighting a young fire dragon when the game crashed. Multiple tries to reopen the game failed, even when I went into Creative Mode so the dragon wouldn't fight me, it still kept crashing. I'll post the crash reports here:

https://pastebin.com/PXqUNbPm
https://pastebin.com/R2u24sih

I'm not a techie, so if the person reading this could please give me a detailed explanation of what exactly is wrong and how to fix it in plain English, that would be really great! This is on my favorite world, and it has a bunch of mods on it, so I'd really like to resume playing normally without it crashing all the time. Thanks!

**Comment by RavenHwk** [comment]
Alright, I figured something out: if I turned off fire, explosion, and smoke animations, turned the Particles setting to "Minimal", and set the gamemode to Creative so the dragon wouldn't attack me, it stopped randomly crashing. Since the fireballs haved worked elsewhere in the world, and even in modded dimensions, it might be worthwhile to note that this fight took place outside of a structure from the mod ChocolateQuest Repoured, whose structures sometimes have special barriers around them that prevent you from breaking blocks withing the barrier, so that might have something to do with it, I don't know. Just thought I'd mention it in case it was any use to you all.

**Comment by TheBv** [comment]
As far as I can tell this was fixed at some point in the 1.12 version but it never got released. #1938
I don't know if I'm allowed to distribute my own builds of this mod. I'll try asking about it but until then if you really want to you could try build the mod yourself. [Here](https://mcforge.readthedocs.io/en/latest/gettingstarted/)'s a resource that could help you a bit. Basically you'd need to download the 1.12 source files from the mods and do the steps described on the page but using the files you've downloaded. I can't promise it'll work since it's quite old but you could try. (Basically step 1 get's replaced with you downloading the source files)
If I'm allowed to distribute my own build of the 1.12 version I'll let you know.

**Comment by RavenHwk** [comment]
Alright, thanks! I'm not the most technically inclined, so I'll wait to hear back first before I try to do any of this complex stuff. I really wish there was a config to make some dragons passive or unless you actively tresspassed on their nest, since I'd rather not fight every single dragon I encounter, especially considering how often they spawn in the Overworld! Even using gosh darn _assault rifles_, they're still a nuisance! Anyway, thanks for the help, I'll try to find a workaround in the meantime (probably just killing the dragon in Creative, so it isn't constantly yeeting fire charges at me while I'm in the area.)

Let me know if there's anything else I can add to help make things easier here!
